File: aula-01/js/Ex09.js
/**
 * Verifica valor de Pi para um total de n termos
 * @param {Number} n quantidade de termos
 * @throws {RangeError} caso n seja menor que 1
 * @returns {Number} retorna o valor calculado de pi
 */
function pi(n) {

    if (n < 1) {
        throw new RangeError("Intervalo de n inválido");
    }
    if (n == null || n == undefined) {
        throw new Error("Entrada não pode ser null ou undefined");
    }
    if (typeof n !== 'number') {
        throw new TypeError("n precisa ser um número");
    }

    let i = 1;
    let s = -1;
    let impar = -1;
    let p = 0;
    while (i <= n) {
        impar = impar + 2;
        s = (-1) * s;
        p = p + 4 * s / impar;
        i = i + 1;
    }

    return p;
}

module.exports = pi;

File: aula-01/java/Ex09.java
public class Ex09 {

    public double pi(int n) {

        if (n < 1) {
            throw new IllegalArgumentException("Intervalo inválido de n");
        }

        double p = 0;
        double s = -1;
        double d = -1;
        int i = 1;

        while (i <= n) {
            d = d + 2;
            s = -1 * s;
            p = p + 4 * s / d;
        }

        return p;
    }
}